src/components/DeeplinkWrapper/index.website.js

```diff
@@ -21,68 +21,76 @@ const propTypes = {
 
         /** Currently logged-in user authToken */
         authToken: PropTypes.string,
+
+        /** The short-lived auth token for navigating to desktop app */
+        shortLivedAuthToken: PropTypes.string,
     }),
 };
 
 const defaultProps = {
     session: {
         email: '',
         authToken: '',
+        shortLivedAuthToken: '',
     },
 };
 
 class DeeplinkWrapper extends PureComponent {
     constructor(props) {
         super(props);
-
-        this.state = {
-            appInstallationCheckStatus: (this.isMacOSWeb() && CONFIG.ENVIRONMENT !== CONST.ENVIRONMENT.DEV)
-                ? CONST.DESKTOP_DEEPLINK_APP_STATE.CHECKING : CONST.DESKTOP_DEEPLINK_APP_STATE.NOT_INSTALLED,
-        };
-        this.focused = true;
+        this.hasPopupBeenOpenedBefore = !this.isMacOSWeb() || CONFIG.ENVIRONMENT === CONST.ENVIRONMENT.DEV;
     }
 
     componentDidMount() {
-        if (!this.isMacOSWeb() || CONFIG.ENVIRONMENT === CONST.ENVIRONMENT.DEV) {
+        if (this.hasPopupBeenOpenedBefore) {
             return;
         }
 
-        window.addEventListener('blur', () => {
-            this.focused = false;
-        });
+        // Since there is no way to know if a previous short-lived authToken is still valid,
+        // any previous short-lived authToken must be cleared out and a new one must be fetched
+        // so that the popup window will only open when we know the short-lived authToken is valid.
+        Session.removeShortLivedAuthToken();
```

Comment:
I don't really get this. Why don't we just not use it when the component mounts? Why does it need to be "cleared out"? It will get replaced with a new token and we will only ever check it once because of the `this.hasPopupBeenOpenedBefore` flag.